engine/glfw/lib/x11/x11_joystick.c

```diff
@@ -79,6 +79,8 @@ struct
 #define JSIOCGBUTTONS  _IOR('j', 0x12, char)  /* get number of buttons (u8) */
 #define JSIOCGNAME(len) _IOC(_IOC_READ, 'j', 0x13, len) /* get identifier string */
 
+static int _glfwJoyConnected[ GLFW_JOYSTICK_LAST + 1 ];
```

Comment:
I opted for this approach since the callback is registered _after_ the glfwInit() is called, and I didn't want to risk disrupting the code flow for all platforms.